### PR TITLE
feat(plugin-meetings): added metrics for actions and failures (USM Flow)

### PR DESCRIPTION
**Pull request by Neeraj-swarnkar**
# COMPLETES #<[ SPARK-293392 ](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-293392) & [SPARK-349919](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-349919)>

## This pull request addresses

Added dropdown selection for conversationURL for Adhoc flowin samples
Added Metrics for - fetchmeeting success, fetchmeeting failure
captcha, password
adhoc meeting failure
Also, Added Tests for same

Below Screenshot from amplitude -

image

image

### To Do

- [x] Testing Password/Captcha Flow
- [x] Show the metrics in amplitude for Password/Captcha Flow
- [x] Testing Adoc Meeting
- [x] Show the metrics for Adoc Meeting inamplitude
- [x] Test Cases need to be written for few failure cases (If Required)

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 1]
```diff
@@ -1001,10 +1012,25 @@ export default class Meeting extends StatelessWebexPlugin {
...
+          {
+            reason: err.message,
+            stack: err.stack
+          }
+        );
         throw (new CaptchaError());
       }
       else {
         this.meetingInfoFailureReason = MEETING_INFO_FAILURE_REASON.OTHER;
+        Metrics.sendBehavioralMetric(
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:1027]: Change this to fetch meeting info failure  , This is the global error catch 

Add the metrics for adhoc failure here

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 2]
```diff
@@ -968,6 +968,10 @@ export default class Meeting extends StatelessWebexPlugin {
         EVENT_TRIGGERS.MEETING_INFO_AVAILABLE
       );
 
+      Metrics.sendBehavioralMetric(
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:971]: move it to verify password

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 3]
```diff
@@ -1001,10 +1012,25 @@ export default class Meeting extends StatelessWebexPlugin {
         }
 
         this.requiredCaptcha = err.captchaInfo;
+
+        Metrics.sendBehavioralMetric(
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:1016]: move this to meetingInfo v2 fetchMeetingInfo function

File: packages/node_modules/@webex/plugin-meetings/src/metrics/constants.js [hunk 4]
```diff
@@ -44,7 +44,15 @@ const BEHAVIORAL_METRICS = {
   ENABLE_BNR_SUCCESS: 'js_sdk_enable_bnr_success',
   ENABLE_BNR_FAILURE: 'js_sdk_enable_bnr_failure',
   DISABLE_BNR_SUCCESS: 'js_sdk_disable_bnr_success',
-  DISABLE_BNR_FAILURE: 'js_sdk_disable_bnr_failure'
+  DISABLE_BNR_FAILURE: 'js_sdk_disable_bnr_failure',
+  FETCH_MEETING_INFO_V1_SUCCESS: 'js_sdk_fetch_meeting_info_v1_success',
+  FETCH_MEETING_INFO_V1_FAILURE: 'js_sdk_fetch_meeting_info_v1_failure',
+  ADOC_MEETING_INIALIZATION_SUCCESS: 'js_sdk_adoc_meeting_initialization_success',
```
Comment (Neeraj-swarnkar) [inline packages/node_modules/@webex/plugin-meetings/src/metrics/constants.js:50]: As per Arun's suggestion, Need to change ''js_sdk_adoc_meeting_initialization_success'' to 'js_sdk_adoc_meeting_success'

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js (hunk 1 above)
Comment (Neeraj-swarnkar) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Done..

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js (hunk 2 above)
Comment (Neeraj-swarnkar) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Done, Moved to verify password

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js (hunk 3 above)
Comment (Neeraj-swarnkar) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Done, Moved all the sendBehavioralMetric from this to fetchMeetingInfo function

File: packages/node_modules/@webex/plugin-meetings/src/metrics/constants.js (hunk 4 above)
Comment (Neeraj-swarnkar) [reply packages/node_modules/@webex/plugin-meetings/src/metrics/constants.js]: Updated.

File: docs/samples/browser-plugin-meetings/index.html [hunk 5]
```diff
@@ -210,6 +210,7 @@ <h2 class="collapsible">Meeting Management</h2>
                         <option value="personID">Person ID</option>
                         <option value="roomID">Room ID</option>
                         <option value="sipURI">SIP URI</option>
+                        <option value="conversationURL">Conversation URL</option>
```
Comment (arun3528) [inline docs/samples/browser-plugin-meetings/index.html:213]: lets change the value to CONVERSATION_URL

Comment (arun3528) [reply docs/samples/browser-plugin-meetings/index.html:213]: remove all the other values except the conversationUrl

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.js [hunk 6]
```diff
@@ -189,18 +205,47 @@ export default class MeetingInfoV2 {
     if (directURI) options.directURI = directURI;
 
     return this.webex.request(options)
+      .then(() => {
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.js:208]: if V1 is present its the new meetinginfo !?

**Review by arun3528 — COMMENTED** [review]
only some changes in the samples app rest looks good

File: docs/samples/browser-plugin-meetings/app.js [hunk 7]
```diff
@@ -323,19 +324,38 @@ function collectMeetings() {
...
   e.preventDefault();
 
-  const {value} = createMeetingDestinationElm;
+  const { value } = createMeetingDestinationElm;
 
+  const type = createMeetingSelectElm.value;
+  console.log("MeetingsManagement#createMeeting type", type);
   console.log('MeetingsManagement#createMeeting', value);
 
-  webex.meetings.create(value)
```
[the remark this answers is not in the document]
Comment (Neeraj-swarnkar) [reply docs/samples/browser-plugin-meetings/app.js:333]: Ok, Sure @arun3528, I just checked in api https://webex.github.io/webex-js-sdk/api/#create
type is optional. let me do the change and update.

File: packages/node_modules/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.js (hunk 6 above)
Comment (Neeraj-swarnkar) [reply packages/node_modules/@webex/plugin-meetings/src/meeting-info/meeting-info-v2.js:208]: @arun3528, This fetches meeting info from the server, correct as discussed earlier, if MEETING_INFO_V1_SUCCESS it's the new meeting info

**Comment by Neeraj-swarnkar** [comment]
> only some changes in the samples app rest looks good
Thanks, @arun3528, Let me make the changes and commit.

File: docs/samples/browser-plugin-meetings/index.html (hunk 5 above)
Comment (Neeraj-swarnkar) [reply docs/samples/browser-plugin-meetings/index.html]: Done, changed the value to CONVERSATION_URL and removed other values.

File: docs/samples/browser-plugin-meetings/app.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (Neeraj-swarnkar) [reply docs/samples/browser-plugin-meetings/app.js:333]: Changes are done, it's working fine. Thanks.

**Comment by Neeraj-swarnkar** [comment]
@arun3528, Can you please re review, if nothing needs to be addressed we can go ahead with approve and merge.

**Review by arun3528 — APPROVED** [review]

**Pull request merged**